itests/fe/service_accounts_test.py

```diff
@@ -27,37 +18,41 @@
     from tests.setup import SetupTest
 
 
-def test_service_account_lifecycle(async_server, browser):  # noqa: F811
-    browser.get(url(async_server, "/groups/user-admins"))
-
-    page = GroupViewPage(browser)
-    page.click_add_service_account_button()
+def test_service_account_lifecycle(tmpdir, setup, browser):
```

Comment:
Not review, just a general question:

What are your thoughts on basing tests on real workflows like this? It's very natural and I think tends to end up reading very nicely, but I've always stayed away from them to prioritize being comprehensive. So here for example, there are a bunch of small corner cases that aren't exercised.

Reply:
I like them. My preferred style is to have one of these tests and then separate tests for the interesting edge cases. The drawback is that the Selenium tests are very slow, so ideally as much of the edge case testing as possible is done at the usecase test level, and then the Selenium test only tests the full workflow happy path and then any UI-specific edge cases.

In general, though, I'm happy to duplicate test coverage to have both edge-case and full workflow tests.